proxy/request.go

```diff
@@ -65,7 +65,7 @@ func (r *request) executeInternal(next bool) {
 		}
 		if r.host == nil {
 			r.done = true
-			r.send(&message.Unavailable{ErrorMessage: "No more hosts available (exhausted query plan)"})
+			r.send(&message.ServerError{ErrorMessage: "Proxy exhausted query plan and there are no more hosts available to try"})
```

Comment:
I really want to use `Unavailable` because that makes more sense, but I don't want to trigger a retry policy to happen in the driver which would be redundant. 

In fact, I'm wonder if it doesn't make sense to use `ServerError` for already retried error (Unavailable, Read Timeout, etc.) to prevent the driver retry policy from acting on the errors. I've created a follow up issue: https://github.com/datastax/cql-proxy/issues/67